core/dbt/config/runtime.py

```diff
@@ -16,17 +16,16 @@
 from dbt.contracts.connection import AdapterRequiredConfig, Credentials
 from dbt.contracts.graph.manifest import ManifestMetadata
 from dbt.contracts.relation import ComponentName
-from dbt.ui import warning_tag
 
 from dbt.contracts.project import Configuration, UserConfig
 from dbt.exceptions import (
     RuntimeException,
     DbtProjectError,
     validator_error_message,
-    warn_or_error,
     raise_compiler_error,
 )
-
+from dbt.events.functions import warn_or_error
+from dbt.events.types import UnsedResourceConfigPath
```

Comment:
very cool PR! caught one typo while reading through
```suggestion
UnusedResourceConfigPath
```

Reply:
Good eye!